src/vm_cpu.rs

```diff
@@ -95,6 +97,139 @@ pub struct VmCpuTrapState {
     pub htval: u64,
     pub htinst: u64,
 }
+#[derive(Default)]
+#[repr(C)]
+struct VmPmuState {
```

Comment:
Is the name bit misleading ? AFAIK, all these states are per vcpu. Just VmPmuState may confuse the readers to be per VM only. But that's me reading rust code :). 

Feel free to ignore my comment if it makes sense to you.

Reply:
I am open to suggestions, but VmCpuVmState might be too verbose?

Reply:
It's fine then.